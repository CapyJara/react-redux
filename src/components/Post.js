import React from 'react';
import PropTypes from 'prop-types';

function Post({ post }) {
  return (
    <>
      <h3>{post.title}</h3>
      <p>{post.postBody}</p>
    </>
  );
}

Post.propTypes = {
  post: PropTypes.shape({
    title: PropTypes.string.isRequired,
    postBody: PropTypes.string.isRequired
  })
};

export default Post;

