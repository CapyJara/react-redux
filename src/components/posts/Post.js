import React from 'react';
import PropTypes from 'prop-types';
import {
  Link
} from 'react-router-dom';

function Post({ post }) {
  console.log(post.id);
  return (
    <Link to={`/post/id=${post.id}`}>
      <h3>{post.title}</h3>
    </Link>
  );
}

Post.propTypes = {
  post: PropTypes.shape({
    title: PropTypes.string.isRequired,
    postBody: PropTypes.string.isRequired
  })
};

export default Post;

