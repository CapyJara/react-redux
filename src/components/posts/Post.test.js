import React from 'react';
import { shallow } from 'enzyme';
import Post from './Post';

describe('Post component', () => {
  it('renders a Post', () => {
    const post = { title: 'myPost', postBody: 'thing i talked about' };
    
    const wrapper = shallow(<Post post={post} />);
    expect(wrapper).toMatchSnapshot();
  });
});
