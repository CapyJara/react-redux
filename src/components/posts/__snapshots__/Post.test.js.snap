// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Post component renders a Post 1`] = `
<Link
  to="/post/id=undefined"
>
  <h3>
    myPost
  </h3>
</Link>
`;
